File: The Deck/Models/entity/Table.go
package entity

import "time"

type Table struct {
	Id        uint      `json:"id"`
	Number    int       `json:"number" gorm:"type:int" validate:"required"`
	Capacity  int       `json:"capacity" gorm:"type:int" validate:"required"`
	Status    string    `json:"status" gorm:"type:varchar(20)" validate:"required"`
	CreatedAt time.Time `json:"created_at" gorm:"autoCreateTime" db:"created_at"`
	UpdatedAt time.Time `json:"updated_at" gorm:"autoUpdateTime" db:"updated_at"`
}

File: frontend/the_deck/lib/Models/Table.dart
class Table {
  final int id;
  final int number;
  final int capacity;
  final int status;

  Table(
      {required this.id,
      required this.number,
      required this.capacity,
      required this.status});

  factory Table.fromJson(Map<String, dynamic> json) {
    return Table(
      id: json['id'],
      number: json['number'],
      capacity: json['capacity'],
      status: json['status'],
    );
  }
}
